llvm/lib/Target/AMDGPU/AMDGPUPreloadKernargHeader.cpp

```diff
@@ -0,0 +1,229 @@
+//===- AMDGPUPreloadKernargHeader.cpp - Preload Kernarg Header ------------===//
```

Comment:
Perhaps dumb question, what's being referred to as Header in this context? Is it similar/the same as the llvm loop terminology of a header?

Reply:
I'm open to suggestions on the name. By header I just mean I'm adding on a preamble to the kernel. Could be prologue, or we could just name it backward compatibility layer or some variation of that.

Reply:
Prolog or preamble would be better